todo/viewsets.py

```diff
@@ -1,141 +1,83 @@
-        data['owner'] = request.user
-
         self.perform_create(serializer)
         headers = self.get_success_headers(serializer.data)
-        todo = get_object_or_404(Todo, name= data['name'])
-        log = Log(
-                user=request.user,
-                name_log = data['name'],
-                action=f'Created object: ' + str(data['name']),
-                timestamp=datetime.now(),
-            )
-        log.save()
         
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
     
     def update(self, request, *args, **kwargs):
         instance = self.get_object()
         serializer = self.get_serializer(instance, data=request.data, partial=True)
         serializer.is_valid(raise_exception=True)
-        old_name = instance.name
         self.perform_update(serializer)
-        data = serializer.validated_data
 
         headers = self.get_success_headers(serializer.data)
 
-        todo = get_object_or_404(Todo, name= data['name'])
-        log = Log(
-                user=request.user,
-                name_log = old_name,
-                action=f'Update object: ' + str(data['name']),
-                timestamp=datetime.now(),
-            )
-        log.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK, headers=headers)
     
-    def destroy(self, request, *args, **kwargs):
-        instance = self.get_object()
-        self.perform_destroy(instance)
-
+    def perform_destroy(self, instance):
         log = Log(
-                user=request.user,
+                user=instance.owner,
                 name_log = instance.name,
                 action=f'Deleted object: ' + str(instance.name),
                 timestamp=datetime.now(),
             )
         log.save()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        instance.delete()
     
     def export_to_excel(self, request, *args, **kwargs):
-        todos = self.get_queryset()
-
-        #Create Workbook
-        wb = Workbook()
-        ws = wb.active
-
-        #Write Header
-        headers = ['ID','Name', 'Description', 'Status', 'Owner', 'Date of Changed']
-        ws.append(headers)
-
-        #write data to worksheet
-        for todo in todos:
-            owner_user = todo.owner
-            username = owner_user.username
-            row = [todo.id, todo.name, todo.description, todo.status, username, todo.date]
-            ws.append(row)
-            
-        file_name = 'todo_export.xlsx'
-        content_type = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
-
-        # Create response object with file attachment
-        response = HttpResponse(content_type=content_type)
-        response['Content-Disposition'] = f'attachment; filename="{file_name}"'
-        # Save workbook to response
-        wb.save(response)
 
-        return response
+        export_function = export_to_excel(self,request)
```

Comment:
ถ้า return value เป็น Response ก็ควรใช้ variable name ว่า `exported_response` นะ แบบนี้มันไม่สื่อ ถ้าอ่านดูจะคิดว่ามันเก็บ function instead of response

แต่ทั้งนี้พี่คิดว่าการทำ response อะ view ควรเป็นคนทำ เพราะงั้นลองเปลี่ยนให้ func คืน wb ออกมาก็พอครับ (next step: แล้ว variable name ควรเป็นไรดี)